packages/next/src/server/web/sandbox/sandbox.ts

```diff
@@ -82,7 +82,8 @@ export const run = withTaggedErrors(async function runWithTaggedErrors(params) {
   const runtime = await getRuntimeContext(params)
   const subreq = params.request.headers[`x-middleware-subrequest`]
   const subrequests = typeof subreq === 'string' ? subreq.split(':') : []
-  if (subrequests.includes(params.name)) {
+  const allowSubrequests = params.request.headers[`x-middleware-allow-subrequests`] === "true"
+  if (subrequests.includes(params.name) && !allowSubrequests) {
```

Comment:
Hi, maybe this could be more specific to dev mode and only allowed there for testing as mentioned in the use case. Also what do you think about `x-dev-subrequests-bypass` for the header and it's only honored if `process.env.NODE_ENV === 'development'`?

cc @javivelasco for thoughts as well

Reply:
We'd be cool with this, also down for a loud log of some form when this path is hit and we can handle educating users on our side if that's a blocker

Reply:
I’d be happy to make that change 👍